server/v2/cometbft/abci.go

```diff
@@ -222,7 +223,10 @@ func (c *Consensus[T]) Query(ctx context.Context, req *abciproto.QueryRequest) (
 func (c *Consensus[T]) InitChain(ctx context.Context, req *abciproto.InitChainRequest) (*abciproto.InitChainResponse, error) {
 	c.logger.Info("InitChain", "initialHeight", req.InitialHeight, "chainID", req.ChainId)
 
-	// store chainID to be used later on in execution
+	// verify chain id
+	if c.chainID != req.ChainId {
+		return nil, fmt.Errorf("chain-id mismatch, got %s, expected %s", req.ChainId, c.chainID)
+	}
 	c.chainID = req.ChainId
```

Comment:
isn't this assignment always ineffectual since the code returns an err when `c.chainID != req.chainId`?